artifactory/usage/reportusage.go

```diff
@@ -67,8 +71,16 @@ func isVersionCompatible(artifactoryVersion string) bool {
 	return version.AtLeast(minArtifactoryVersion)
 }
 
-func reportUsageToJson(productId, commandName string) ([]byte, error) {
+func reportUsageToJson(productId, commandName string, attributes ...ReportUsageAttribute) ([]byte, error) {
 	featureInfo := feature{FeatureId: commandName}
+	if len(attributes) > 0 {
+		featureInfo.Attributes = make(map[string]string, 0)
+		for _, attribute := range attributes {
+			if attribute != (ReportUsageAttribute{}) {
```

Comment:
Since you're checking for emptiness multiple times in the code, I suggest creating a method `isEmpty` for readability. Also, there is no meaning in values without a key. Therefore I think it is enough to create and use isEmpty method as follows:
```go
func (rua *ReportUsageAttribute) isEmpty() bool {
	return rua.AttributeName == ""
}
```

Reply:
done